Get all info in job containers

karmelp opened the pull request:
almost done showing all data from json

karmelp committed:
show company name & position

Nii. Põhimõtteliselt sain, aga kuna see on list of array, siis suursin igalepoole esimese objeti valued saada. Uurin edasi.

karmelp committed:
Update script.js

karmelp committed:
i did it

karmelp committed:
done

karmelp committed:
almost done showing data from json

karmelp committed:
Update script.js

Kui toolse välja toodud pole, siis kuidas ma teen nii, et see ei tekitaks tühja koha jaoks konteinerit?
Minu loogika käis läbi if job.tools is not empty then do this, else do none, aga ma ei suutnud seda teostada. 😁

Teine küss. Kui list array objekt on omakorda array, siis how do i separate them? Praegu võtab tool ja language köik ühte patta.
Jälle. Katsetasin mingeid viise, aga ei saanud hakkama.

Ja ma ei tea kuidas postedat, contract ja location vahele täppi teha 😂 proovisin jsis svg circleit teha 🤓😁 aga ma ei suutnud möelda sellele,

hhabakuk commented:
minu arvutis su kood ei tööta, jälle mingi failide laadimise jama, ma kunagi proovin süveneda, milles asi võib olla. 
aga praegu puusalt: 

**Kui toolse välja toodud pole, siis kuidas ma teen nii, et see ei tekitaks tühja koha jaoks konteinerit?**

ma ei näe hetkel, mida sa proovinud oled ja millisel koodireal see lisakonteiner tekkima hakkab. oleneb meetodist, mida sa kuidas kasutad. Näita seda koodi, mis selle lisakonteineri tekitab. 

**Kui list array objekt on omakorda array, siis how do i separate them? Praegu võtab tool ja language köik ühte patta.**
Pakun, et jutt käib sellest koodireast:
```
// jobDetailsEl.innerHTML = `${job.languages} ${job.tools}`;
```

siin pole mingit vahet, kus sees on "job". Käitu `job.languages` nagu arrayga ikka.  
```
const languages = job.languages
// kuva iga keele kohta mingi element.
```

**Ja ma ei tea kuidas postedat, contract ja location vahele täppi teha**

Ma arvan, et kõige lihtsam oleks see teha centerdot tähemärgiga, nt copypasta siit: https://www.compart.com/en/unicode/U+2022

Teine variant oleks cssis ring teha. Alati kui sul on mingi lihtne disainielement vaja teha, eelista cssi mingile svg/jpg-le -> palju optimaalsem lahend brauserile,

karmelp committed:
proovisin võtta languaget lahti, no ei oska :D see katsetus jäi viimaseks praegu

karmelp committed:
teine lahendus, mida google täis on, n midagi sellist, aga see ka mul tööle ei hakka :D

karmelp committed:
Update script.js

karmelp committed:
featured tag loogika, mis on vale :)

karmelp committed:
panin sama loogikaga new tagi japanin need 2 tagi õigetesse kohtadesse ka

karmelp committed:
Update script.js

karmelp committed:
põhimõtteliselt mo tools loogika töötab, ainult et ma ei suuda välja möelda, kuidas mitte midagi kuvada

karmelp committed:
siis ma hakkasin mõtlema, et äkki ma saan sellisel juhul lihtsalt selle toolsi arrayst eemaldada, aga seegi ei önnestunud :D

karmelp committed:
loogika loksus uute ja featured tagidega paika

karmelp committed:
lahendasin tools ära - sain kõvasti naerda :D

karmelp committed:
nokitsen languagesi lahti vötmise kallal nüüd

karmelp commented on script.js:
```diff
@@ -174,10 +174,17 @@ function getLevel(job) {
-    languagesTagEl.innerHTML = `${job.languages}`;
-    return languagesTagEl;
+
+    for (const [language, value] of Object.entries(job.languages)) {
```
selle koodiga saan hetkel sellise tulemuse

<img width="529" alt="Screenshot 2022-10-14 at 15 38 18" src="https://user-images.githubusercontent.com/101389533/195849616-b24c1e9d-3625-40a0-9953-33873ff82815.png">

tundub köige lähemal, mis ma olnud olen, aga samas mötlen, et kui arvesse võtta sinu hommikust vihjet, siis kui mapida läbi languages key ja siis teha foreach igaühele ja luua iga kohta see tag... hmmmmm......

karmelp commented on script.js:
```diff
@@ -174,10 +174,17 @@ function getLevel(job) {
-    languagesTagEl.innerHTML = `${job.languages}`;
-    return languagesTagEl;
+
+    for (const [language, value] of Object.entries(job.languages)) {
```
selle koodiga saan hetkel sellise tulemuse

<img width="152" alt="Screenshot 2022-10-14 at 15 43 23" src="https://user-images.githubusercontent.com/101389533/195849772-625f9faa-0552-4c39-b01e-8d01e368ce24.png">

tundub köige lähemal, mis ma olnud olen, aga samas mötlen, et kui arvesse võtta sinu hommikust vihjet, siis kui mapida läbi languages key ja siis teha foreach igaühele ja luua iga kohta see tag... hmmmmm......

hhabakuk replied:
miks sa ei saa teha `job.languages.map`? copy-paste mulle, mida tagastab `console.log(job.languages)`

hhabakuk replied:
ja mida see pilt tähendab :D ?

hhabakuk replied:
aaaaa sain aru pildist

karmelp replied:
kui sa vaatad edasi mo pushe, siis näed praegust versiooni :D

hhabakuk replied:
Sul on lihtsalt üks array, arrayga läbikäimine ei sõltu kuidagi sellest, kust see array tuli. See on samasugune array nagu todo listis oli sul tasks. Logi välja  `console.log(languagesForJobListing) ` ja sa näed, see tagastab lihstalt stringidest array. 

Kui sul oleks vaja kuvada iga array elemendi kohta LI element ehk sul on array 

```
const array = ['esimene', 'teine', 'kolmas']
```


ja htmli peab tekkima

```
<li>esimene</li>
<li>teine</li>
<li>kolmas</li>
```

kas sa endiselt alustaksid kirjutades `for (const [order] of Object.entries(array))` ? 
kuidas sa kuvaks sellised elemendid?

karmelp committed:
praegu jälle sellise mõtte juures languagesitega

karmelp committed:
okei, nüüd ma saan esimese keele igaühelt

karmelp committed:
job.languages.map nüüd :)

karmelp committed:
a pole ka, mingi error oli mul

karmelp committed:
map

karmelp committed:
languages

ma jään sulle vastuse võlgu. ma ausalt kõike proovinud juba ja no miski ei tööta, see for...of looop oli köige lähemal, sest vähemalt ta töi esimese keele välja, teised landused toovad ainult undefined vöi vabsjee errori :D

hhabakuk commented on script.js:
```diff
@@ -174,13 +174,22 @@ function getLevel(job) {
 function getLanguages(job) {
-
-    job.languages.map((language) => {
+    job.languages.forEach(language => {
```
jaa, see töötab, sa kuvad küll keeli (React, React + Sass)
<img width="414" alt="image" src="https://user-images.githubusercontent.com/13555621/195876008-35dddfb4-8ce7-4299-a355-a3416e6158e9.png">

hhabakuk replied:
a ei... ota ma vaatan :D

hhabakuk replied:
igatahes foreach töötab sul korrektselt, console logiga tuleb õige info
```
function getLanguages(job) {
  job.languages.forEach((language) => {
    const languagesTagEl = document.createElement("div");
    languagesTagEl.className = "jobDeatilTag";
    languagesTagEl.innerHTML = `${language}`;
    
    console.log("language", language);   SEE ON KORRAS!
     
    return languagesTagEl;
  });
```

hhabakuk commented on script.js:
```diff
@@ -0,0 +1,236 @@
+    // võta level tag getLevel funktsioonist
+    const levelTagEl = getLevel(job);
+    // võta keelte tag getLanguages funktsioonist
+    const languagesTagEl = getLanguages(job)
```
kuna `getLanguages` leiab õiged keeled ilusti üles, tulin uurima samm edasi -> kes kutsub välja getLanguages?

siin on näha, et kui teha `console.log(languagesTagEl)` tagastatakse `undefined`. Ehk su probleem on selles, et `getLanguages` leiab küll õiged keeled üles, aga ta ei tagasta mitte midagi. Sa lihtsalt käid arrayd läbi, aga ei anna tagasi elementi, kus oleks sees kõik languaged. 

lahend on midagi sellist: 

```
function getLanguages(job) {
  const wrapperEl = document.createElement("div"); TEED MINGI KASTI, KUHU KOGUDA KEELI
  
  
  job.languages.forEach((language) => {       KÄID KÕIK KEELED LÄBI
    const languagesTagEl = document.createElement("div");  TEED IGA KEELE KOHTA ELEMENDI
    languagesTagEl.className = "jobDeatilTag";
    languagesTagEl.innerHTML = `${language}`;  IGA ELEMENDI TEKSTIKS PANED KEELE

    wrapperEl.append(languagesTagEl); TOPID KEELE ELEMENDI KEELEKOGUMISKASTI
  });
  
  return wrapperEl; TAGASTAD KEELEKOGUMISKASTI
  }
  ```

karmelp replied:
aaapppiiiiiiii. jälle mingi täiesti jabur asi :D Aga no ma tänan :D

karmelp committed:
kõik data nähtaval
